internal/pkg/cli/local_run.go

```diff
@@ -0,0 +1,243 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/aws/copilot-cli/internal/pkg/aws/identity"
+	"github.com/aws/copilot-cli/internal/pkg/aws/sessions"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/deploy"
+	"github.com/aws/copilot-cli/internal/pkg/describe"
+	"github.com/aws/copilot-cli/internal/pkg/term/color"
+	"github.com/aws/copilot-cli/internal/pkg/term/log"
+	"github.com/aws/copilot-cli/internal/pkg/term/prompt"
+	"github.com/aws/copilot-cli/internal/pkg/version"
+	"github.com/aws/copilot-cli/internal/pkg/workspace"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type localRunVars struct {
+	wkldName string
+	appName  string
+	envName  string
+}
+
+type localRunOpts struct {
+	localRunVars
+
+	deployedWkld       []string
+	wkldDeployedToEnvs map[string][]string
+	store              store
+	ws                 wsWlDirReader
+	prompt             prompter
+	deployStore        deployedEnvironmentLister
+	envVersionGetter   func(string) (versionGetter, error)
+}
+
+func newLocalRunOpts(vars localRunVars) (*localRunOpts, error) {
+	sessProvider := sessions.ImmutableProvider(sessions.UserAgentExtras("local run"))
+	defaultSess, err := sessProvider.Default()
+	if err != nil {
+		return nil, err
+	}
+
+	store := config.NewSSMStore(identity.New(defaultSess), ssm.New(defaultSess), aws.StringValue(defaultSess.Config.Region))
+	deployStore, err := deploy.NewStore(sessProvider, store)
+	if err != nil {
+		return nil, err
+	}
+
+	ws, err := workspace.Use(afero.NewOsFs())
+	if err != nil {
+		return nil, err
+	}
+
+	opts := &localRunOpts{
+		localRunVars: vars,
+
+		deployedWkld:       []string{},
+		wkldDeployedToEnvs: make(map[string][]string),
+		prompt:             prompt.New(),
+		store:              store,
+		ws:                 ws,
+		deployStore:        deployStore,
+	}
+	opts.envVersionGetter = func(envName string) (versionGetter, error) {
+		envDescriber, err := describe.NewEnvDescriber(describe.NewEnvDescriberConfig{
+			App:         opts.appName,
+			Env:         envName,
+			ConfigStore: opts.store,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("new environment compatibility checker: %v", err)
```

Comment:
nit we don't usually wrap the error returned by initiators
```suggestion
			return nil, err
```